### PR TITLE
qt5 5.6.2

### PR DESCRIPTION
**Pull request by ilovezfs**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
Fails to build for me.  [Logs here](https://gist.github.com/rleigh-codelibre/f52e5ce2a4dd26d3a7247aaab96d79c1).  However, note that while the build "fails" there's nothing obvious in the logs--it appears to have built to completion; maybe something in the formula after that point?

**Comment by rleigh-codelibre** [comment]
Looks like this is https://github.com/Homebrew/homebrew-core/issues/5391#issuecomment-251606873

**Comment by ilovezfs** [comment]
@rleigh-codelibre already fixed that a while ago, if you re-pull :)

File: Formula/qt5.rb [hunk 1]
```diff
@@ -27,28 +26,6 @@ class Qt5 < Formula
     sha256 "73d33dd2563c39542844c276a7bd43463f2974fde141e7afeb3057168adbe606" => :mavericks
   end
 
-  # Restore `.pc` files for framework-based build of Qt 5 on OS X. This
```
Comment (DomT4) [inline Formula/qt5.rb]: I think getting rid of this would still be a mistake, FWIW.

Comment (MikeMcQuaid) [inline Formula/qt5.rb]: @DomT4 Can you elaborate a bit more?

Comment (DomT4) [inline Formula/qt5.rb]: I suspect it's still going to be very breaking, especially as Homebrew moves stuff from `qt` to `qt5`. The Qt ecosystem isn't exactly known for its rapid response to upstream changes. Up to y'all though, I'm no longer around really, just noticed this and thought I'd add the comment.

**Comment by rleigh-codelibre** [comment]
@ilovezfs Thanks.  Tested again on 10.12 and it builds and links successfully, and my code builds and runs against it without problems.

**Comment by ilovezfs** [comment]
qca problem is fixed by a revision bump.

qjackctl problem is caused by opportunistic linking to `portaudio` during bottling and is fixed by adding the dependency or explicitly excluding it during configure.

Both are unrelated to this PR and can be fixed before, during, or after it without consequence.

File: Formula/qt5.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/qt5.rb]: @DomT4 thanks for chiming in. For the formulae mentioned in the comment, the situation currently is

- wireshark: upstream fixed it
- mkvtoolnix: upstream provides the configure option `--without-qt-pkg-config` and then checks the environment variables `QT_CFLAGS` and `QT_LIBS`
- poppler: upstream checks the environment variables `POPPLER_QT5_CFLAGS`, `POPPLER_QT5_LIBS`, `POPPLER_QT5_TEST_CFLAGS`, and `POPPLER_QT5_TEST_LIBS` as an alternative to pkg-config.

So if the patch is removed, these are the required diffs:
```
diff --git a/Formula/mkvtoolnix.rb b/Formula/mkvtoolnix.rb
index cd2e296..17ed39d 100644
--- a/Formula/mkvtoolnix.rb
+++ b/Formula/mkvtoolnix.rb
@@ -64,6 +64,10 @@ class Mkvtoolnix < Formula
     if build.with?("qt5")
       qt5 = Formula["qt5"]
 
+      ENV["QT_CFLAGS"] = "-I#{qt5.opt_include} -I#{qt5.opt_include}/QtNetwork -I#{qt5.opt_include}/QtWidgets -I#{qt5.opt_include}/QtGui -I#{qt5.opt_include}/QtCore"
+      ENV["QT_LIBS"] = "-F#{qt5.opt_lib} -framework QtNetwork -framework QtWidgets -framework QtGui -framework QtCore"
+
+      args << "--without-qt-pkg-config"
       args << "--with-moc=#{qt5.opt_bin}/moc"
       args << "--with-uic=#{qt5.opt_bin}/uic"
       args << "--with-rcc=#{qt5.opt_bin}/rcc"
```
```
diff --git a/Formula/poppler.rb b/Formula/poppler.rb
index a405768..f860c15 100644
--- a/Formula/poppler.rb
+++ b/Formula/poppler.rb
@@ -53,6 +53,13 @@ class Poppler < Formula
     ]
 
     if build.with? "qt5"
+      qt5 = Formula["qt5"]
+
+      ENV["POPPLER_QT5_CFLAGS"] = "-I#{qt5.opt_include} -I#{qt5.opt_include}/QtXml -I#{qt5.opt_include}/QtWidgets -I#{qt5.opt_include}/QtGui -I#{qt5.opt_include}/QtCore"
+      ENV["POPPLER_QT5_LIBS"] = "-F#{qt5.opt_lib} -framework QtXml -framework QtWidgets -framework QtGui -framework QtCore"
+      ENV["POPPLER_QT5_TEST_CFLAGS"] = "-I#{qt5.opt_include} -I#{qt5.opt_include}/QtTest -I#{qt5.opt_include}/QtCore"
+      ENV["POPPLER_QT5_TEST_LIBS"] = "-F#{qt5.opt_lib} -framework QtTest -framework QtCore"
+
       args << "--enable-poppler-qt5"
     else
       args << "--disable-poppler-qt5"
```

There are probably similar changes that would be needed in other taps too. If upstream is not going to reverse its decision to remove the pkg-config files, it seems at some point we need to bite the bullet and make the requisite changes (and do the corresponding upstream reporting) in formulae that are affected.

Comment (MikeMcQuaid) [inline Formula/qt5.rb]: > it seems at some point we need to bite the bullet and make the requisite changes (and do the corresponding upstream reporting) in formulae that are affected.

Agreed. We should stick to the upstream version wherever possible.

**Comment by ilovezfs** [comment]
qca fixed by https://github.com/Homebrew/homebrew-core/pull/5860
qjackctl fixed by https://github.com/Homebrew/homebrew-core/pull/4836

File: Formula/qt5.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/qt5.rb]: poppler qt5 option fixed in https://github.com/Homebrew/homebrew-core/pull/5864
mkvtoolnix qt5 option fixed in https://github.com/Homebrew/homebrew-core/pull/5865

**Pull request closed without merging**